hera_mc/geo_handling.py

```diff
@@ -19,298 +21,397 @@
+            counter = 0
+            for connection in connected_station.all():
+                connection.gps2Time()
+                if cm_utils._is_active(active_date, connection.start_date, connection.stop_date):
+                    station_connected = connection.downstream_part+':'+connection.down_part_rev
+                    counter += 1
+                else:
+                    station_connected = False
+            if counter > 1:
+                print("Error:  more than one active connection for", station_name)
+                station_connected = False
+        return station_connected
+
+
+    def find_station_of_antenna(self, antenna, query_date):
+        """
+        checks to see at which station an antenna is located
+
+        Returns False or the active station_name (must be an active station for the query_date)
+
+        Parameters:
+        ------------
+
+        antenna:  antenna number as float (why?), int, or string. If needed, it prepends the 'A'
+        query_date:  is the astropy Time for contemporary antenna
+        """
+
+        if type(antenna) == float or type(antenna) == int or antenna[0] != 'A':
+            antenna = 'A' + str(antenna).strip('0')
+        connected_antenna = self.session.query(part_connect.Connections).filter( (part_connect.Connections.downstream_part == antenna)  &
+                                                                                 (query_date.gps >= part_connect.Connections.start_gpstime) )
+        ctr = 0
+        for conn in connected_antenna:
+            if conn.stop_gpstime is None or query_date.gps <= conn.stop_gpstime:
+                antenna_connected = copy.copy(conn)
+                ctr+=1
+        if ctr == 0:
+            antenna_connected = False
+        elif ctr > 1:
+            raise ValueError('More than one active connection between station and antenna')
+        return antenna_connected.upstream_part
+
+
+    def get_location(self, to_find, query_date, show_location=False, verbosity='m'):
+        """
+        Return the location of station_name or antenna_number as contained in args.locate.
+        This accepts the fact that antennas are sort of stations, even though they are parts
+
+        Parameters:
+        ------------
+        to_find:  station names to find (must be a list)
+        query_date:  astropy Time for contemporary antenna
+        show_location:   if True, it will print the information
+        verbosity:  sets the verbosity of the print
+        """
+        self.get_station_types(add_stations=True)
+        found_location = []
+        for L in to_find:
+            station_name = False
+            try:
+                antenna_number = int(L)
+                station_name = self.find_station_of_antenna(antenna_number, query_date)
+            except ValueError:
+                station_name = L.upper()
+            found_it = False
+            if station_name:
```

Comment:
Can this if statement ever fail? If so, it looks like the returned list will just be missing entries for those stations. I think this might be very confusing for a user, because they wouldn't know which ones were missing. I strongly suggest adding a None to the returned list in the location of the station that wasn't found.